pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -26,6 +26,7 @@ in
 , doBenchmark ? false
 , doHoogle ? true
 , doHaddockQuickjump ? doHoogle && lib.versionAtLeast ghc.version "8.6"
+, doInstallDist ? false
```

Comment:
On first reading, I thought this flag was going to be about `sdist`.
I've never understood why the directory is called `dist`. It is not for distributable files, although I can see a distant connection.

So for the output name, which I believe we should try to standardize beyond just haskellPackages, I think we should find a distinct name that isn't as distorted, so as not to disturb users who expect "distributable" files in such an output. https://github.com/NixOS/nixpkgs/pull/167670 uses a derivation name suffix `incrementalBuildArtifacts` rather than an output name, but it is rather distended. Perhaps that name could be distilled into something shorter.

(I hope my use of dist- words was not too distracting.)

Reply:
I think `incrementalBuildArtifacts` can definitely be shortened to at least `buildArtifacts` (since they don't necessarily need to be used for incremental builds) and it's still an accurate name.  The question is if that could be shortened even further to just `build` or `artifacts`.

I would personally try to get away with shortening that to just `artifacts` since I believe that name has the right connotation: "artifacts" tends to mean historic byproducts that were ancillary to the final outcome (in both a programming context and non-programming context).  Also, the `build` in `buildArtifacts` seems superfluous since everything produced by a Nix derivation is part of a build of some sort.

Reply:
I'm used to artifacts referring to what's typically already in the outputs, rather than the intermediate files that do not need to be exposed for a package to be useful.
How about `intermediate`?

> historic byproducts that were ancillary to the final outcome

Isn't this true of all software, no matter how complete?

> the `build` in `buildArtifacts` seems superfluous [...]

It may refer to `buildPhase`, but that's not obvious.